src/plugins/intel_cpu/src/transformations/tpp/x64/pass/lowered/brgemm_tpp_blocking.cpp

```diff
@@ -32,19 +32,9 @@ std::shared_ptr<snippets::lowered::pass::PassBase> BrgemmTPPBlocking::SetBrgemmB
     return !other || ov::is_type<SetBrgemmBeta>(other) ? std::make_shared<SetBrgemmBeta>() : nullptr;
 }
 
-std::tuple<size_t, size_t, size_t> BrgemmTPPBlocking::get_blocking_params(const ov::snippets::lowered::ExpressionPtr& brgemm_expr) {
-    const auto& in_0_desc = brgemm_expr->get_input_port_descriptor(0);
-    const auto& in_1_desc = brgemm_expr->get_input_port_descriptor(1);
-    const auto& out_desc = brgemm_expr->get_output_port_descriptor(0);
-
-    const auto& in_0_planar_dims = get_planar_vdims(in_0_desc->get_shape(), in_0_desc->get_layout());
-    const auto& in_1_planar_dims = get_planar_vdims(in_1_desc->get_shape(), in_1_desc->get_layout());
-    const auto& out_preordered_dims = get_preordered_vdims(out_desc->get_shape(), out_desc->get_layout());
-
-    const auto& m = *++out_preordered_dims.rbegin();
-    const auto& n = *out_preordered_dims.rbegin();
-    const auto& k = *in_0_planar_dims.rbegin();
-    OPENVINO_ASSERT(k == *++in_1_planar_dims.rbegin(), "Brgemm input descriptors have different K dimension value.");
+std::tuple<size_t, size_t, size_t> BrgemmTPPBlocking::get_blocking_params(const ov::snippets::lowered::ExpressionPtr& brgemm_expr) const {
+    size_t m, n, k;
+    std::tie(m, n, k) = get_brgemm_dimensions(brgemm_expr);
```

Comment:
BrgemmTPP blocking heuristic is as straightforward as it can possibly be, so why don't we use the base implementation even here? For example, do we need to substitute full_dim_value here?
Can we use the incremental initialization here as well?
1) get default blocking
2) if (m_blocking == get_full_dim_value()) m_blocking = m;

Reply:
I removed overrided method from tpp and call `get_projected_subtensor` in the corresponding emitters. Looks like it's enough

UPD: Unfortunately it leaded to accuracy degradations, It looks like there are some places which should be updated also. So I implemented your suggestion. Thanks